maven-enforcer-rules/src/main/java/io/helidon/build/maven/enforcer/rules/HelidonDependenciesRule.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.build.maven.enforcer.rules;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.enforcer.rule.api.AbstractEnforcerRule;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * This rule will apply the same basic "eclipse transformer" checks, but applied as an enforcer rule.
+ */
+// see https://maven.apache.org/enforcer/enforcer-api/writing-a-custom-rule.html
+// see https://projects.eclipse.org/projects/technology.transformer
+@Named("helidonDependenciesRule") // rule name - must start from lowercase character
+@SuppressWarnings("unused")
+public class HelidonDependenciesRule extends AbstractEnforcerRule {
+    /**
+     * An appropriate value for {@link #namespace} (value={@value}).
+     */
+    static final String JAVAX = "javax";
+    /**
+     * An appropriate value for {@link #namespace} (value={@value}).
+     */
+    static final String JAKARTA = "jakarta";
+
+    @Inject
+    private MavenProject project;
+
+    @Inject
+    private MavenSession session;
+
+    /**
+     * Rule parameter as list of gav regular expressions. Reflectively injected from pom.xml
+     */
+    // @Inject
```

Comment:
Commented code

Reply:
intended to relay that this is injected by maven.  i plan to keep this that way.